### ISSUE TITLE
fix: lightgbm placement clusters basis units into one subrange

### ISSUE THREAD

**Issue by ChrisW09** [OpenTabular/PreTab#10]
## Summary

Every consumer of `placement_strategy="lightgbm"` (PLE, the four feature maps, and the
target-aware splines) places its basis units in a small subregion of the feature range
instead of spreading them where the target signal is. The gain ranking that motivates
using LightGBM over CART is also discarded entirely.

`BaseLocationSelector._enforce_spacing` filters candidates with
`point - spaced[-1] >= min_distance`, which is only correct for an **ascending** list.
`LightGBMLocationSelector._ordered_candidates` deliberately returns candidates in
**gain-descending** order — the base-class docstring at `pretab/core/selectors.py:121-126`
states this is the contract. Any candidate whose threshold is lower than the last kept one
produces a negative difference, fails the test, and is dropped, so the filter degenerates
into "walk the gain-ordered list and keep an increasing subsequence".

## Reproduction

```python
import numpy as np
from pretab.core.selectors import LightGBMLocationSelector

rng = np.random.default_rng(0)
x = rng.uniform(0, 10, size=2000)
y = np.sin(x) + 0.05 * rng.normal(size=2000)

sel = LightGBMLocationSelector()
cands, _ = sel._ordered_candidates(x.reshape(-1, 1), y, "regression")
print("candidates (gain-descending):", len(cands))
print("ascending?", cands == sorted(cands))
print("after _enforce_spacing:", np.round(sel._enforce_spacing(cands, x.reshape(-1, 1)), 2))
print("select(min=3, max=8):", np.round(sel.select(x, y, task="regression", min_count=3, max_count=8), 2))
```

```
candidates (gain-descending): 184
ascending? False
after _enforce_spacing: [6.27 6.44 6.59 9.14 9.29 9.52 9.68 9.82]
select(min=3, max=8): [6.27 6.44 6.59 9.14 9.29 9.52 9.68 9.82]
```

184 candidates collapse to 8, all inside `[6.27, 9.82]` of a `[0, 10]` range. A spacing
filter that compares against every kept location (rather than only the previous one) keeps
65 of the 184.

The same defect through the public API:

```python
import numpy as np, pandas as pd
from pretab import Preprocessor

rng = np.random.default_rng(0)
x = rng.uniform(0, 10, size=2000)
y = np.sin(x) + 0.05 * rng.normal(size=2000)
df = pd.DataFrame({"x": x})

for ps in ("cart", "lightgbm"):
    pre = Preprocessor(numerical_method="ple", placement_strategy=ps,
                       target_aware=True, output_dim=8).fit(df, y)
    th = pre.column_transformer_.named_transformers_["num_x"].named_steps["ple"].thresholds_[0]
    print(f"{ps:9s} PLE thresholds:", np.round(th, 2))
```

```
cart      PLE thresholds: [-0.3   0.15  0.26  0.38  0.73  0.82  0.91]
lightgbm  PLE thresholds: [0.26 0.29 0.32 0.82 0.87 0.9  0.94]
```

## Expected

Locations spread across the feature range, ranked by cumulative gain — comparable coverage
to the `cart` strategy, which is what the `lightgbm` docstring promises ("tends to find
informative locations that a single tree can miss").

## Actual

All seven thresholds land in the upper 40% of the scaled range; the lower 60% of the
feature collapses into a single PLE bin.

## Root cause

Two compounding defects:

1. `pretab/core/selectors.py:135-148` — `_enforce_spacing` assumes ascending input.
2. `pretab/core/selectors.py:362-363` — `LightGBMLocationSelector._trim_over_max` does
   `points[:max_count]` on the assumption that gain order survived `_enforce_spacing`. It
   does not: the surviving list is ascending, so this keeps the *lowest* thresholds rather
   than the *highest-gain* ones.

For comparison, the true top-8 by gain on the data above is
`[2.835, 5.903, 5.996, 6.067, 6.269, 6.318, 6.444, 6.591]` — quite different from what the
selector returns.

## Suggested fix

Make `_enforce_spacing` order-independent so it honours the documented per-selector
ordering, e.g. keep a candidate only when it is at least `min_distance` from **every**
already-kept location:

```python
def _enforce_spacing(self, split_points, x):
    if len(split_points) <= 1:
        return split_points
    min_distance = self.min_location_spacing * float(x.max() - x.min())
    spaced = [split_points[0]]
    for point in split_points[1:]:
        if all(abs(point - kept) >= min_distance for kept in spaced):
            spaced.append(point)
    return spaced
```

That preserves CART behaviour exactly (its candidates are already sorted, so comparing
against all kept locations is equivalent to comparing against the previous one) while
letting the LightGBM path keep its gain ordering, which then makes `points[:max_count]`
correct again.

Worth adding a regression test asserting that `lightgbm` locations span a comparable
fraction of the feature range to `cart` locations.

## Impact

Affects `Preprocessor(placement_strategy="lightgbm")` for `numerical_method` in
`ple`, `rbf`, `relu`, `sigmoid`, `tanh`, `bspline`, `mspline`, `ispline`, `cubicspline`,
`naturalspline`, and the `LightGBMKnotSelector` / `LightGBMLocationSelector` public classes.
Output is silently degraded — no error, no warning.

## Environment

- pretab 0.1.0 (`main` @ 51c3043)
- Python 3.11.15, numpy 2.4.6, pandas 2.3.3, scikit-learn 1.9.0, scipy 1.17.1, lightgbm 4.7.0
- macOS (darwin 25.5.0)

**Comment by mkumar73** [comment]
@ChrisW09  - thank you for reporting the issue. Unfortunately, due to merge conflict we couldn't accept the PR. However, we have included the fix with v1.0.0 release.
